src/pykeen/triples/triples_factory.py

```diff
@@ -593,6 +664,120 @@ def split(
+        """Create a semi-inductive split.
+
+        In a semi-inductive split, we first split the entities into training and evaluation entities.
+        The training graph is then composed of all triples involving only training entities.
+        The evaluation graphs are built by looking at the triples that involve exactly one training
+        and one evaluation entity.
+
+        :param ratios:
+            The *entity* split ratio(s).
+        :param random_state:
+            The random state used to shuffle and split the triples.
+
+        :return:
+            A partition of triples, which are split (approximately) according to the ratios, stored TriplesFactory's
+            which share everything else with this root triples factory.
+
+        .. seealso::
+            - [ali2021]_
+        """
+        # Make new triples factories for each group
+        return [
+            self.clone_and_exchange_triples(
+                mapped_triples=triples,
+                # do not explicitly create inverse triples for testing; this is handled by the evaluation code
+                create_inverse_triples=None if i == 0 else False,
+            )
+            for i, triples in enumerate(
+                split_semi_inductive(mapped_triples=self.mapped_triples, ratios=ratios, random_state=random_state)
+            )
+        ]
+
+    def split_fully_inductive(
+        self,
+        entity_split_train_ratio: float = 0.5,
+        evaluation_triples_ratios: float | Sequence[float] = 0.8,
+        random_state: TorchRandomHint = None,
+    ) -> list[Self]:
+        """Create a fully inductive split.
+
+        In a fully inductive split, we first split the entities into two disjoint sets:
+        training entities and inference entities. We use the induced subgraph of the training entities for training.
+        The triples of the inference graph are then further split into inference triples and evaluation triples.
+
+        :param entity_split_train_ratio:
+            The ratio of entities to use for the training part. The remainder will be used for the
+            inference/evaluation graph.
+        :param evaluation_triples_ratios:
+            The split ratio for the inference graph split.
+        :param random_state:
+            The random state used to shuffle and split the triples.
+
+        :return:
+            A (transductive) training triples factory, the inductive inference triples factory,
+            as well as the evaluation triples factories.
+        """
+        training, inference, *evaluation = split_fully_inductive(
+            mapped_triples=self.mapped_triples,
+            entity_split_train_ratio=entity_split_train_ratio,
+            evaluation_triples_ratios=evaluation_triples_ratios,
+            random_state=random_state,
+        )
+        # TODO: do we need to adjust the entity map of training? (or inference?)
+        training_tf = self.clone_and_exchange_triples(mapped_triples=training).condense()
```

Comment:
if you condense these both seperately, will they have entity mappings that are out of sync?

Reply:
Yes, and for the fully inductive setting that *should* be fine. In this setting, we actually have two separate graphs, training and inference, where evaluatin always happens in the evaluation graph. That's also why we have separate representations for those.